### PR TITLE
Update the README to be in sync with erlef/setup-elixir

### PR DESCRIPTION
**Pull request by starbelly**
- changed README.md to mirror setup-elixir for consistency where
 possible
 - add CONTRIBUTING.md and CODE_OF_CONDUCT.md
 
 So, let's dialog what we expect things to look like. I think what's in here mostly makes sense, two things aren't clear yet and/or were not relevant to setup-erlang vs setup-elixir

1. Do we say rebar or rebar3? I believe rebar3 will go in OTP at some point as `rebar`. Feels right, but also feels technically wrong. 
2. There are no matchers yet, it isn't clear what we want/need yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,79 @@
...
+on: push
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-erlang@v1
+        with:
+          otp-version: '22.2'
```
Comment (paulo-ferraz-oliveira) [inline README.md:40]: I'd named this `erlang-version` but am OK to change it. I guess elixir called it that because it would otherwise be weird to have `erlang-version` alongside `otp-version`. I usually refer to the combo as Erlang/OTP, but yeah, whatever you prefer.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,79 @@
...
+          rebar3-version: '3.14.2'
```
Comment (paulo-ferraz-oliveira) [inline README.md:41]: This is already outdated 😄

File: README.md (hunk 1 above)
Comment (starbelly) [reply README.md:40]: I have no strong feelings on this. erlang-version is fine.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good
+faith may face temporary or permanent repercussions as determined by other
+members of the project's leadership.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
+available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html
```
Comment (paulo-ferraz-oliveira) [inline CODE_OF_CONDUCT.md:72]: FYI, version 2.0's already out. 😄

File: README.md [hunk 4]
```diff
@@ -1,4 +1,79 @@
...
-:boom:
+[![](https://github.com/erlef/setup-erlang/workflows/Test/badge.svg)](https://github.com/erlef/setup-erlang/actions)
+[![](https://github.com/erlef/setup-erlang/workflows/Licensed/badge.svg)](https://github.com/erlef/setup-erlang/actions)
+
+This action sets up an Erlang/OTP environment for use in a GitHub Actions
+workflow by:
+
+- Installing OTP
+- Installing rebar3
+
```
Comment (paulo-ferraz-oliveira) [inline README.md:11]: It's missing Hex.

And I'd rephrase it to "optionally install `rebar3`". It doesn't have to be optional, though. I'm just taking example from `setup-elixir`, where it is. Hex to.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,79 @@
...
+**Note** Currently, this action currently only supports Actions' `ubuntu-` runtimes.
```
Comment (paulo-ferraz-oliveira) [inline README.md:12]:
```suggestion
**Note** Currently, this action only supports Actions' `ubuntu-` runtimes.
```

File: README.md [hunk 6]
```diff
@@ -1,4 +1,79 @@
...
-Set up your GitHub Actions workflow with Erlang/OTP
 
-:boom:
+[![](https://github.com/erlef/setup-erlang/workflows/Test/badge.svg)](https://github.com/erlef/setup-erlang/actions)
+[![](https://github.com/erlef/setup-erlang/workflows/Licensed/badge.svg)](https://github.com/erlef/setup-erlang/actions)
+
+This action sets up an Erlang/OTP environment for use in a GitHub Actions
+workflow by:
+
+- Installing OTP
```
Comment (paulo-ferraz-oliveira) [inline README.md:9]:
```suggestion
- Installing Erlang/OTP
```

File: README.md [hunk 7]
```diff
@@ -1,4 +1,79 @@
...
+- Installing rebar3
+
+**Note** Currently, this action currently only supports Actions' `ubuntu-` runtimes.
+
+## Usage
+
+See [action.yml](action.yml).
+
+**Note** The OTP release version specification is [relatively
```
Comment (paulo-ferraz-oliveira) [inline README.md:18]:
```suggestion
**Note** The Erlang/OTP release version specification is [relatively
```

File: README.md [hunk 8]
```diff
@@ -1,4 +1,79 @@
...
+complex](http://erlang.org/doc/system_principles/versions.html#version-scheme).
+For best results, we recommend specifying exact OTP and rebar3 versions.
```
Comment (paulo-ferraz-oliveira) [inline README.md:20]:
```suggestion
For best results, we recommend specifying exact Erlang/OTP and `rebar3` versions.
```

File: README.md [hunk 9]
```diff
@@ -1,4 +1,79 @@
...
+However, values like `22.x` are also accepted, and we attempt to resolve them
+according to semantic versioning rules.
+
+Additionally, it is recommended that one specifies OTP and rebar3 versions
```
Comment (paulo-ferraz-oliveira) [inline README.md:24]:
```suggestion
Additionally, it is recommended that one specifies Erlang/OTP and `rebar3` versions
```

File: README.md [hunk 10]
```diff
@@ -1,4 +1,79 @@
...
+
+### Matrix example
+
+```yaml
+on: push
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: OTP ${{matrix.otp}} / Rebar3 ${{matrix.rebar}}
```
Comment (paulo-ferraz-oliveira) [inline README.md:54]:
```suggestion
    name: Erlang/OTP ${{matrix.otp}} / rebar3 ${{matrix.rebar}}
```

File: README.md [hunk 11]
```diff
@@ -1,4 +1,79 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-erlang@v1
+        with:
+          otp-version: '22.2'
+          rebar3-version: '3.14.2'
+      - run: rebar3 deps
```
Comment (paulo-ferraz-oliveira) [inline README.md:42]: This one's probably extraneous, since `rebar3` pull and compiles and everything by default.

File: README.md [hunk 12]
```diff
@@ -1,4 +1,79 @@
...
+    strategy:
+      matrix:
+        otp: ['20.3', '21.3', '22.2']
+        rebar3: ['3.14.1', '3.14.3']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-erlang@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          rebar3: ${{matrix.rebar3}}
```
Comment (paulo-ferraz-oliveira) [inline README.md:64]:
```suggestion
          rebar3-version: ${{matrix.rebar3}}
```

File: README.md [hunk 13]
```diff
@@ -1,4 +1,79 @@
...
+      - run: rebar3 update
```
Comment (paulo-ferraz-oliveira) [inline README.md:65]: We could remove this to keep the example simpler. What say you?

File: README.md (hunk 11 above)
Comment (starbelly) [reply README.md:42]: Yeah, that's right 🤔

File: README.md (hunk 13 above)
Comment (starbelly) [reply README.md:65]: Yeah makes sense to me

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (starbelly) [reply CODE_OF_CONDUCT.md:72]: We can copy/pasta that.

File: README.md (hunk 4 above)
Comment (starbelly) [reply README.md:11]: Can you clarify on missing hex?

File: README.md (hunk 1 above)
Comment (paulo-ferraz-oliveira) [reply README.md:40]: Cool; change it?

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (paulo-ferraz-oliveira) [reply CODE_OF_CONDUCT.md:72]: I'm hoping it didn't break the interface 😄

File: README.md (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply README.md:11]: From `setup-elixir`:
```
  install-hex:
    description: Whether to install Hex
    default: true
```

Comment (paulo-ferraz-oliveira) [reply README.md:11]: And this is what that means `mix local.hex --force`. I'm not sure there's an equivalent for `rebar3`, though.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (starbelly) [reply CODE_OF_CONDUCT.md:72]: You think this version is needed for the action to work? Not sure I follow 🤔

File: README.md (hunk 4 above)
Comment (starbelly) [reply README.md:11]: Don't need it. rebar3 ships with hex core, we don't recommend using rebar3 hex in ci, though people do.

**Comment by paulo-ferraz-oliveira** [comment]
Would you mind copying LICENSE.md over also?

File: README.md (hunk 1 above)
Comment (starbelly) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply README.md:11]: One less worry. That's good!

File: README.md (hunk 11 above)
Comment (paulo-ferraz-oliveira) [inline README.md]:
```suggestion
```

File: README.md (hunk 13 above)
Comment (paulo-ferraz-oliveira) [inline README.md]:
```suggestion
```

**Review by paulo-ferraz-oliveira — APPROVED** [review]
